core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -105,11 +110,7 @@ public void run() {
           }, Configuration.getInt(PropertyKey.USER_HEARTBEAT_INTERVAL_MS),
           Configuration.getInt(PropertyKey.USER_HEARTBEAT_INTERVAL_MS), TimeUnit.MILLISECONDS);
 
-      try {
-        sessionHeartbeat();
-      } catch (InterruptedException e) {
-        throw Throwables.propagate(e);
-      }
+      NUM_ACTIVE_SESSIONS.incrementAndGet();
```

Comment:
Why do we increment here even though the heartbeat has not be successful? Also, what's the issue with putting the heartbeat in the constructor?